reco_utils/dataset/amazon_reviews.py

```diff
@@ -233,6 +237,83 @@ def _data_generating(input_file, train_file, valid_file, test_file, min_sequence
             cate_list.append(category)
             dt_list.append(date_time)
 
+def _data_generating_no_history_expanding(input_file, train_file, valid_file, test_file, min_sequence=1):
```

Comment:
Can you help to add some commentary to this two function? _data_generating and _data_generating_no_history_expanding,  for easiler understanding.

Reply:
added